### PR TITLE
Exposing `Time` as `icu::calendar::Time`

### LINKED ISSUE #4335: Re-export Time in icu::calendar?
There is `icu::calendar::Date` and `icu::calendar::DateTime` but not `icu::calendar::Time`.

We do have a `Time` type but it is in `icu::calendar::types::Time` which is much less discoverable.

I think we should re-export `Time`.

CC @Manishearth

### PR DESCRIPTION
**Pull request by robertbastian**
Fixes #4335

### CODE REVIEW

**Review by Manishearth — DISMISSED** [review]

File: ffi/capi/src/time.rs [hunk 1]
```diff
@@ -6,18 +6,22 @@
...
-    use icu_calendar::types::Time;
+    use icu_calendar::Time;
 
     use crate::errors::ffi::ICU4XError;
 
     #[diplomat::opaque]
     /// An ICU4X Time object representing a time in terms of hour, minute, second, nanosecond
-    #[diplomat::rust_link(icu::calendar::types::Time, Struct)]
+    #[diplomat::rust_link(icu::calendar::Time, Struct)]
+    #[diplomat::rust_link(icu::calendar::types::Time, Struct, hidden)]
```
Comment (sffc) [inline ffi/capi/src/time.rs:16]: Question: is this how we decided to deal with re-exports? Seems cumbersome. Why did you need to duplicate the rust_links in some cases but not others?

Comment (Manishearth) [reply ffi/capi/src/time.rs:16]: We should make the tool smart about reexports, really.

Comment (sffc) [reply ffi/capi/src/time.rs:16]: I lean toward keeping reexports in allowlist.rs instead of spreading the problem out through the bridge code.

Comment (Manishearth) [reply ffi/capi/src/time.rs:16]: I agree.

**Review by sffc — CHANGES_REQUESTED** [review]
^

**Review by sffc — DISMISSED** [review]

**Review by sffc — APPROVED** [review]
